src/cljs/athens/views/block_page.cljs

```diff
@@ -87,20 +96,23 @@
          [:span {:style {:color "gray"}}
           [breadcrumbs-list {:style {:font-size "1.2rem"}}
            (doall
-             (for [{:keys [node/title block/uid block/string]} parents]
-               ^{:key uid}
-               [breadcrumb {:key (str "breadcrumb-" uid) :on-click #(navigate-uid uid)}
-                (or title string)]))]]
+            (for [{:keys [node/title block/uid block/string]} parents]
+              ^{:key uid}
+              [breadcrumb {:key (str "breadcrumb-" uid) :on-click #(navigate-uid uid)}
+               (or title string)]))]]
 
          ;; Header
          [:h1 (use-style title-style {:data-uid uid :class "block-header"})
-          [autosize/textarea
-           {:id          (str "editable-uid-" uid)
-            :value       (:string/local @state)
-            :class       (when (= editing-uid uid) "is-editing")
-            :auto-focus  true
-            :on-key-down (fn [e] (node-page/handle-key-down e uid state nil))
-            :on-change   (fn [e] (block-page-change e uid state))}]
+          (let [diff? (when (not= (:string/local @state)
+                                  (:string/previous @state)) true)]
```

Comment:
Does this variable need to be bound outside the textarea, or can it be done from within `transact-string`? This value only needs to be computed on `blur` at the moment, but is instead computed on any change to the component including each key press. What if we put this logic, including `when diff?` inside `transact-string`.

Also, most event handlers in the Athens codebase have the event name as part of their name. It might make more sense to call `transact-string` something like `handle-blur` or `handle-textarea-blur` or `handle-header-blur`.

Reply:
Separate point: `when...true` seems like not a great practice too. `not=` will return a boolean true or false. Thoughts?

Reply:
Thought I responded to this, but must have run before hitting enter.
1) Good point. Thoughts initially were to build `transact-string` to be as abstract as possible so could possibly be used elsewhere. Current fix should be better. 
1a) IMHO naming things is one of the hardest things to do in software 😂 My understanding of best practice is to try to avoid clunking up the names of functions with unnecessary or redundant verbs.. The re-frame documentation touches on it [here](https://day8.github.io/re-frame/EffectfulHandlers/#pattern-structure) . Most of my experience with this comes from having to be disciplined about avoiding the temptation to call every component `render-the-thing` . However, I see the benefit of having consistency among the codebase so I'll leave it up to you. 
2) Apologies, bad habit- now you know my secret!